### PR TITLE
Reversing of array

### LINKED ISSUE #498: Title: reversing of Array
### Description

Description:
Greetings!
I want to add problem of reversing  array with two methods->
1.with the help of another array  in which firstly we will copy elements of our main array  in helping array from the back side  then again copying elements of helping array in our main array with taking O(n) time-complexity.
2. with the help of two pointers first will iterate from left to right while second will iterate right to left until left<right and swapping the array[first pointer] with array[second pointer]  with  taking O(n) time-complexity.

please assign this issue to me as I am a contributor in GSSOC'22.

### Domain

Data Structure

### Type of Contribution

Addition

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) & [Code of conduct](https://github.com/Lakhankumawat/LearnCPP/blob/main/.github/CODE_OF_CONDUCT.md) of LearnCPP.

### PR DESCRIPTION
**Pull request by i-am-aisha**
->REVERSING OF ARRAY
create a file reversing of array.cpp in  which I code 2 methods of reversing any array -->
1. reverse1--->reversing any array with the help of auxilliary array in this method first we will copy the elements of first array in auxilliary array in reverse order and then again copying the elements of auxilliary array to main array this will reverse our main array.
2. reverse2-->with the help of two pointers high and low, low will iterate from left to right while high will iterate form right to left until low<high and swapping the high index element with low index element this will reverse our main array.
I code above 2 methods with proper comments and mention  time complexity which is O(n) and space complexity which is O(n) in reverse1 and O(1) in reverse2.

## Related Issue
close #498

## Proposed Changes
- Create a reversing of array.cpp file
- update README.md  accordingly and mention about my file in README.md with some test cases and discuss about the algorithm of  both methods of reversing any array i.e. reverse1 and reverse2.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

## Output Screenshots
code image 1
code-image1

code image 2
code-image2 (2)

input:
input (2)

output:
output

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.


///------------------Description -----------------------///

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.



///------------------Description -----------------------///

Let me show you how to check the check list , there should be no space inside brackets like [x]
 `[x]`
- [x] Checklist
- [ ] Or you can just check them from outside without doing any of these
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [ ] 🌟 ed the repo

1. Starr the repo if you haven't !
2. Have you watched this [video](https://youtu.be/kIFwqp9B7pQ) ?


///----------------------PR README --------------------///


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by i-am-aisha** [comment]
conflict resolved👍

File: A-Array/Reversing of array.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//space-complexity-->O(1) -->Constant
+}
+
+
+int main()
+{
+    //Declaring and initializing the array
+    int arr[] = { 1, 2, 3, 4, 5 };  
+  
+    //Calculating the size of the array
```
Comment (siddhi-244) [inline A-Array/Reversing of array.cpp:52]: Try to take array as input !

Comment (i-am-aisha) [reply A-Array/Reversing of array.cpp]: change the code as well as PR for the same👍

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
You haven't added the reverse array approach in readme.

**Comment by i-am-aisha** [comment]
readme updated👍

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Algorithm should be the steps of program in simple words And not the code. Please see to it !

**Comment by i-am-aisha** [comment]
updated👍

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
You have deleted the entire readme file and then copy pasted the code. Please see to it !

**Comment by i-am-aisha** [comment]
Updated👍

**Comment by i-am-aisha** [comment]
I updated everything you said but now  what should I do now like what I have to do to correct it?
can you help me please?

**Comment by siddhi-244** [comment]
You can make a fresh new pr then ! And just add your content instead of deleting other's content

**Comment by i-am-aisha** [comment]
you mean separate pr  only for readme of reversing-of-array?

**Comment by siddhi-244** [comment]
close this pr and make a new one with same commits ! Just dont delete the content of whole readme file !

**Pull request closed without merging**